flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManager.java

```diff
@@ -116,8 +116,8 @@ void start(
      * @param initialSlotReport for the new task manager
      * @param totalResourceProfile for the new task manager
      * @param defaultSlotResourceProfile for the new task manager
-     * @return True if the task manager has not been registered before and is registered
-     *     successfully; otherwise false
+     * @return True if the task manager has been registered before or is registered successfully;
+     *     otherwise false
```

Comment:
The new semantic is against intuition and hard to understand. Moreover, it leads to duplicated calls of `ResourceManager#onWorkerRegistered`.

Alternatively, we can introduce a enum type for the return value. E.g.,
```
enum RegistrationResult {
    SUCCESS, // task manager has not been registered before and is registered successfully
    IGNORED, // task manager has been registered before and is ignored
    REJECTED, // task manager is rejected and should be disconnected
}
```